apps/andi/lib/andi/input_schemas/struct_tools.ex

```diff
@@ -0,0 +1,101 @@
+defmodule Andi.InputSchemas.StructTools do
+  @moduledoc false
+  def to_map(%_{} = struct) do
+    struct
+    |> struct_to_map()
+    |> to_map()
+  end
+
+  def to_map(map) when is_map(map) do
+    map
+    |> Enum.map(fn {k, v} ->
+      {k, to_map(v)}
+    end)
+    |> Map.new()
+  end
+
+  def to_map(value) when is_binary(value) do
+    value
+  end
+
+  def to_map(list) when is_list(list) do
+    list
+    |> Enum.map(&to_map/1)
+  end
+
+  def to_map(value) do
+    value
+  end
+
+  def preload(nil, _fields), do: nil
+
+  def preload(list, fields) when is_list(list) do
+    Enum.map(list, &preload(&1, fields))
+  end
+
+  def preload(%struct_type{} = struct, fields) do
```

Comment:
:eyes: I want to know whats going on here, I really do, but I don't

Reply:
This is basically loading all of the associations of a struct in ecto based on the type of each of those associations. It's pretty gross but necessary. I can work on cleaning this up a little